lang/src/org/partiql/lang/prettyprint/QueryPrettyPrinter.kt

```diff
@@ -785,29 +783,7 @@ class QueryPrettyPrinter {
             is PartiqlAst.Type.NullType -> sb.append("NULL")
             is PartiqlAst.Type.AnyType -> sb.append("ANY")
             is PartiqlAst.Type.BagType -> sb.append("BAG")
-            is PartiqlAst.Type.ScalarType -> when (TYPE_ALIAS_TO_SCALAR_TYPE[node.alias.text]) {
-                BuiltInScalarType.BOOLEAN -> sb.append("BOOLEAN")
-                BuiltInScalarType.INTEGER -> sb.append("INT")
-                BuiltInScalarType.SMALLINT -> sb.append("SMALLINT")
-                BuiltInScalarType.INTEGER4 -> sb.append("INT4")
-                BuiltInScalarType.INTEGER8 -> sb.append("INT8")
-                BuiltInScalarType.FLOAT -> sb.append("FLOAT")
-                BuiltInScalarType.REAL -> sb.append("REAL")
-                BuiltInScalarType.DOUBLE_PRECISION -> sb.append("DOUBLE_PRECISION")
-                BuiltInScalarType.DECIMAL -> sb.append("DECIMAL")
-                BuiltInScalarType.NUMERIC -> sb.append("NUMERIC")
-                BuiltInScalarType.TIMESTAMP -> sb.append("TIMESTAMP")
-                BuiltInScalarType.CHARACTER -> sb.append("CHAR")
-                BuiltInScalarType.CHARACTER_VARYING -> sb.append("VARCHAR")
-                BuiltInScalarType.STRING -> sb.append("STRING")
-                BuiltInScalarType.SYMBOL -> sb.append("SYMBOL")
-                BuiltInScalarType.BLOB -> sb.append("BLOB")
-                BuiltInScalarType.CLOB -> sb.append("CLOB")
-                BuiltInScalarType.DATE -> sb.append("DATE")
-                BuiltInScalarType.TIME -> sb.append("TIME")
-                BuiltInScalarType.TIME_WITH_TIME_ZONE -> sb.append("TIME WITH TIME ZONE")
-                else -> error("Unrecognized scalar type ID")
-            }
+            is PartiqlAst.Type.ScalarType -> sb.append(node.alias.text.toUpperCase())
```

Comment:
You need to replace "_" with " "

Reply:
Actually, we don't if a user really wants "_" in the original type name or not. e.g. If I want to define a new type, whose type name is `MY_TYPE`, the transformation to `MY TYPE` can cause parsing error. 

Probably we can consider multiple [identifier](https://github.com/partiql/partiql-lang-kotlin/blob/a9e5a4ef090ffa73caa2d85771c4322acbed5e4d/partiql-grammar/src/main/antlr/org/partiql/grammar/PartiQLTokens.g4#L352)s with spaces in between as valid type name, and require no underscore in type names lexically?

Reply:
If we can make such change in partiql grammar, then lexically a space corresponds to an underscore defined here for type names. Then, the problem is lexically where do we put type parameters for those types with name to be multiple identifiers. Probably `TIME (2) WITH TIME ZONE` is a good example.

Reply:
This might be a good design, but as of now, the use cases are still kind of ambiguous. So I think it's fine to just keep this line as is?

Reply:
If you do not replace "_" with " ", then the pretty printer will produce unparse-able PartiQL. Because the current design replaces " " with "_" in the lexer, you need to undo this when printing PartiQL text back.

For example,

```sql
--original
SELECT CAST(x AS character varying) FROM Foo
```

The AST is something like
```
query(
  select(
    project(
      cast(
         id(x)
        type(character_varying)
      )
    )
  )
  from(
    scan(
      id(Foo)
    )
)
```

Then pretty printed never removes the "_" inserted by the " ".
```sql
--pretty
SELECT CAST(x AS character_varying) FROM Foo
```

I posted this question elsewhere, but why are we inserted "_" anyways? This seems like a hold-over from the handwritten lexer's multi-lexeme tables. So now if a custom explicitly wants "_" then we can't distinguish between a "_" inserted by the customer or inserted by the lexer. So why insert them in the first place?


Then when you pretty

Reply:
I agree with you that "_" insertion is unnecessary.